src/packagedcode/plugin_package.py

```diff
@@ -328,7 +328,17 @@ def create_package_and_deps(codebase, package_adder=add_to_package, strip_root=F
         strip_root=strip_root,
         **kwargs
     )
-    codebase.attributes.packages.extend(pkg.to_dict() for pkg in packages)
+
+    package_mappings = []
+    for package in packages:
+        for detection in package.license_detections:
+            populate_matches_with_path(
+                matches=detection["matches"],
+                path=package.datafile_paths[0],
```

Comment:
Why 0?

Reply:
This was a side-effect of not being able to determine where the license detection was among datafiles, now we've moved this to Package creation so this is not the case anymore.